### ISSUE TITLE
Compiler errors with TS 2.6.1

### ISSUE THREAD

**Issue by cdupuis** [atomist/automation-client-ts#119]
```
src/operations/common/gitHubRepoLoader.ts(14,5): error TS2322: Type '(repoId: GitHubRepoRef) => Promise<GitProject>' is not assignable to type 'RepoLoader<GitProject>'.
  Types of parameters 'repoId' and 'repoId' are incompatible.
    Type 'RepoRef' is not assignable to type 'GitHubRepoRef'.
      Property 'apiBase' is missing in type 'RepoRef'.
src/operations/edit/editAll.ts(38,26): error TS2345: Argument of type 'ProjectEditor<undefined, EditResult<Project>>' is not assignable to parameter of type 'ProjectEditor<P, EditResult<Project>>'.
  Types of parameters 'params' and 'params' are incompatible.
    Type 'P' is not assignable to type 'undefined'.
src/operations/generate/UniversalSeed.ts(20,14): error TS2415: Class 'UniversalSeed' incorrectly extends base class 'SeedDrivenGenerator'.
  Types of property 'projectEditor' are incompatible.
    Type '(ctx: HandlerContext, params: this) => ProjectEditor<undefined, EditResult<Project>>' is not assignable to type '(ctx: HandlerContext, params: this) => AnyProjectEditor<this>'.
      Type 'ProjectEditor<undefined, EditResult<Project>>' is not assignable to type 'AnyProjectEditor<this>'.
        Type 'ProjectEditor<undefined, EditResult<Project>>' is not assignable to type 'SimpleProjectEditor<this>'.
          Types of parameters 'params' and 'params' are incompatible.
            Type 'this' is not assignable to type 'undefined'.
              Type 'UniversalSeed' is not assignable to type 'undefined'.
src/operations/generate/generatorUtils.ts(54,55): error TS2345: Argument of type 'AnyProjectEditor<undefined>' is not assignable to parameter of type 'SimpleProjectEditor<object> | ProjectEditor<object, EditResult<Project>>'.
  Type 'ProjectEditor<undefined, EditResult<Project>>' is not assignable to type 'SimpleProjectEditor<object> | ProjectEditor<object, EditResult<Project>>'.
    Type 'ProjectEditor<undefined, EditResult<Project>>' is not assignable to type 'ProjectEditor<object, EditResult<Project>>'.
      Types of parameters 'params' and 'params' are incompatible.
        Type 'object' is not assignable to type 'undefined'.
src/server/BuildableAutomationServer.ts(100,77): error TS2345: Argument of type 'HandleCommand<P>' is not assignable to parameter of type 'HandleCommand<P | HandleCommand<P>>'.
  Types of property 'handle' are incompatible.
    Type 'OnCommand<P>' is not assignable to type 'OnCommand<P | HandleCommand<P>>'.
      Types of parameters 'parameters' and 'parameters' are incompatible.
        Type 'P | HandleCommand<P>' is not assignable to type 'P'.
          Type 'HandleCommand<P>' is not assignable to type 'P'.
test/operations/generate/AbstractGeneratorTest.ts(12,7): error TS2415: Class 'CustomStartSeed' incorrectly extends base class 'AbstractGenerator'.
  Types of property 'projectEditor' are incompatible.
    Type '(ctx: HandlerContext, params: this) => AnyProjectEditor<undefined>' is not assignable to type '(ctx: HandlerContext, params: this) => AnyProjectEditor<this>'.
      Type 'AnyProjectEditor<undefined>' is not assignable to type 'AnyProjectEditor<this>'.
        Type 'ProjectEditor<undefined, EditResult<Project>>' is not assignable to type 'AnyProjectEditor<this>'.
          Type 'ProjectEditor<undefined, EditResult<Project>>' is not assignable to type 'SimpleProjectEditor<this>'.
            Types of parameters 'params' and 'params' are incompatible.
              Type 'this' is not assignable to type 'undefined'.
                Type 'CustomStartSeed' is not assignable to type 'undefined'.
test/operations/review/ReviewerCommandSupport.ts(10,15): error TS2415: Class 'MyReviewer' incorrectly extends base class 'ReviewerCommandSupport<ReviewResult<{}>, ProjectReview>'.
  Types of property 'projectReviewer' are incompatible.
    Type '(context: HandlerContext) => ProjectReviewer<undefined, ProjectReview>' is not assignable to type '(context: HandlerContext) => ProjectReviewer<this, ProjectReview>'.
      Type 'ProjectReviewer<undefined, ProjectReview>' is not assignable to type 'ProjectReviewer<this, ProjectReview>'.
        Types of parameters 'params' and 'params' are incompatible.
          Type 'this' is not assignable to type 'undefined'.
            Type 'MyReviewer' is not assignable to type 'undefined'.
test/project/git/GitProjectTest.ts(70,19): error TS2345: Argument of type '(r: ActionResult<GitCommandGitProject>) => void' is not assignable to parameter of type '(value: ActionResult<GitProject>) => void | PromiseLike<void>'.
  Types of parameters 'r' and 'value' are incompatible.
    Type 'ActionResult<GitProject>' is not assignable to type 'ActionResult<GitCommandGitProject>'.
      Type 'GitProject' is not assignable to type 'GitCommandGitProject'.
        Property 'credentials' is missing in type 'GitProject'.
```

**Comment by cdupuis** [comment]
To create this problem, rev the version of TS in `package.json` to `2.6.1` and run `npm install`.

**Comment by johnsonr** [comment]
2.6.1 appears to tighten up parameters handling considerable. This is non-trivial. Deferring for now.

**Comment by cdupuis** [comment]
This becoming crippling now. We are way behind on TS versions. Can this be fixed cc @johnsonr @jessitron?

**Comment by cdupuis** [comment]
Fixed in #279
